### ISSUE TITLE
[Clarification] How to properly setup a Reverse Deployer for Nemesis/Pandora? + Consider adding Refresh button besides Un/Deploy button

### ISSUE THREAD

**Issue by Heisenborgar** [limo-app/limo#73]
Hello, how do you properly setup a Reverse Deployer so that when you run Nemesis for instance, it will display all files it generated. Is the Reverse Deployer capable of like, separately displaying a tool's generated files, like for example, Nemesis generates 1.txt, while Pandora generates abc.txt file seperately.

My current setup is having a Reverse Deployer pointed to Skyrim Data's folder, but it shows other files generated such as from mods like Powerofthree's Tweaks' INI file. What I wanted is to show what Nemesis or Pandora generated, so I can delete them when I have to due to bugs or having to clear cache. 

Will you also consider like a Refresh button? Some files doesn't show up until you Un/Deploy them.

**Comment by limo-app** [comment]
> Hello, how do you properly setup a Reverse Deployer so that when you run Nemesis for instance, it will display all files it generated. Is the Reverse Deployer capable of like, separately displaying a tool's generated files, like for example, Nemesis generates 1.txt, while Pandora generates abc.txt file seperately.

This is currently not possible, since reverse deployers do not associate files with tools. Retroactively doing this is difficult and would require knowing details about the tools in question, which does not generalize well.
Detecting which file has been generated by a specific tool is in general only possible if you:
1. Associate one or more reverse deployers to every tool that adds/ modifies files
2. Update the files list before the tool is launched
3. Run the tool
5. After the tool exits: Update the list again and associate all new files with the tool

I will add a feature that does this in the future.

> Will you also consider like a Refresh button? Some files doesn't show up until you Un/Deploy them.

This should hopefully not be necessary with the feature mentioned above. The refreshing happens on the deploy action of a reverse deployer, which is why new files only show up afterwards.

**Comment by Heisenborgar** [comment]
Thank you. In the meantime I just undeploy and generally Nemesis/Pandora and other generated files are left behind

**Comment by poperigby** [comment]
In the meantime, you can set the output directory of Pandora to a Limo mod with the -o flag.

**Comment by davozebi** [comment]
I can't get Pandora to recognize my deployed mods. @Heisenborgar @poperigby Did you run into this issue?

**Comment by Heisenborgar** [comment]
Take a look at #29 and see if we have the same problem

**Comment by davozebi** [comment]
It seems I'm not able to execute Pandora outside the prefix that Skyrim is installed and I can't use hard links because of how Flatpaks work.

**Comment by poperigby** [comment]
This is my configuration for Pandora:

![Screenshot_2024-12-21_16-40-42-914580377](https://github.com/user-attachments/assets/5931fd8f-a358-403b-8435-5aef7e43fcea)

**Comment by davozebi** [comment]
> This is my configuration for Pandora:
> 
> ![Screenshot_2024-12-21_16-40-42-914580377](https://private-user-images.githubusercontent.com/20866468/397957786-5931fd8f-a358-403b-8435-5aef7e43fcea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4Mjg0MjUsIm5iZiI6MTczNDgyODEyNSwicGF0aCI6Ii8yMDg2NjQ2OC8zOTc5NTc3ODYtNTkzMWZkOGYtYTM1OC00MDNiLTg0MzUtNWFlZjdlNDNmY2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDAwNDIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1YTdjODY0MGUwMmE4YjdjZTIwMGI0MTMwMWQ0M2Y3NjlkYmQwOTZmZmRmODU5NTUxZDhiMGViYjUwNWEzZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EFi54oRR6zF6mP1bmbW947GEHs6xfjLhbc75tUY9LPo)

Are you deploying your mods via sym links or hard links?

**Comment by poperigby** [comment]
Hard links

**Comment by davozebi** [comment]
No worries! Some solutions I have in mind:
1. Manually install Pandora/Nemesis mods to Data directory
2. Write a script to automate 1.
3. Integrate the same functionality for Pandora/Nemesis mods that happens when SKSE is installed through Limo except the deployers would both be Data but one would be with sym links and the other with CoW. I wonder how difficult it would be to set this up.

**Comment by limo-app** [comment]
> How difficult it would be to integrate 3?

Having multiple normal deployers (i.e. non LOOT deployers) point at the same target would is not really intended and would require some changes. It might be easier to add a new Deployer for your setup that points to the pandora directory. Limo automatically splits mods that would affect multiple target directories. This means that the Nemesis/ Pandora directory would be automatically added to a new mod managed by your new deployer.
In short:
1. Add a new deployer targeting the Pandora directory and set it to copy deployment
2. When installing mods, always add them to the *Data* deployer
3. Limo will split off the Pandora directory from all new mods and add them to the *Pandora* deployer

**Comment by davozebi** [comment]
Thanks for the explanation! I may write a Python script that recursively goes through the sym linked `Nemesis_Engine` directory and have it copy the Targets directly into the `<working directory of Pandora installation>/Nemesis_Engine`.
